New Blob UI! [TG Port]

Arkatos1 opened the pull request:
**What does this PR do:**
Ports Blob UI from /tg/station13, credits to them.

Blobs now have new UI elements, designated to help them use their abilities, so they dont have to clumsily navigate Blob tab to use them. 

Also adds a new verb for them - *Blob Help*. *Blob Help* is reorganized introduction to playing Blob, explaining what they are, and how to play. This introduction is displayed every time a Blob spawns, but now players can read it again, if they missed it or did not have time to read it.

Finally, I moved all the Blob HUD code from the other_mobs.dm to their own new file called blob_overmind.dm, so it can be better worked with in the future.

This PR got major update, please read details below in the comments. Otherwise this PR is ready for the final review.

Tested locally without any issue.

**Images of sprite/map changes (IF APPLICABLE):**
![NewBlobUI](https://user-images.githubusercontent.com/43862960/54941473-369ea500-4f2d-11e9-8907-557dde34076c.png)

**Changelog:**
:cl: Arkatos
add: Added new Blob UI and icons, ported from /tg/station13
add: Added new *Blob Help* verb, designated to help out new players playing as Blob
add: Added descriptions for each Blob reagent
add: Added complementary colors for each Blob reagent
tweak: Blob marker now uses complementary color of Blob reagent as it's own color
tweak: Blob shortcuts updated. Blob shortcuts are now:  Click = Expand Blob | CTRL Click = Create Shield Blob | Middle Mouse Click = Rally Spores | Alt Click = Remove Blob
/:cl:

MrMagolor commented:
There should be info about upgrading strong blobs to reflective.

Arkatos1 commented:
@MrMagolor There actually is, the third sentence in Shield Blobs category.

MrMagolor commented:
> @MrMagolor There actually is, the third sentence in Shield Blobs category.

Oh well, then, this looks great.

Bookdrizzt commented:
This will be a huge improvement to blob rounds and blob midround. Thanks!

Arkatos1 commented:
I have added several improvements.

1) I have added description to each Blob reagent, and added that description to *Blob Help* and when Blob rerolls its chemical. The description contains text about advantages and features of the reageant. Taken from our Paradise wiki.

2) Each Blob reagent now has complementary color, which is basically lighter version of their primary color. Complementary color is now used as a color for a Blob marker. 

3) I have added left click as a shortcut for expanding Blob, and added new shortcut for removing Blob tiles. All shortcuts were reshuffled to have blob controls parity with other codebases.

I am adding updated image of how Blob and *Blob help* looks now: 
![FinalBlobUI](https://user-images.githubusercontent.com/43862960/55030746-45f51f80-500d-11e9-9147-9bf198a28ee6.png)

All changes were tested, and no issues were found. This PR is ready for the final review.

Bookdrizzt commented:
I see an issue. Some people use trackpads not mouses, and can’t middle-click.

Arkatos1 commented:
@Bookdrizzt Good point, but these folks will simply have to use the "classic" way of playing as Blob - via pressing verbs in Blob tab, which is virtually unchanged, and contain all neccessary verbs in it. These are merely shortcuts.

Fox-McCloud commented on code/game/gamemodes/blob/overmind.dm:
```diff
@@ -31,6 +31,7 @@
 	if(blob_core)
 		blob_core.adjustcolors(blob_reagent_datum.color)
 
+	color = blob_reagent_datum.complementary_color
```
I don't get the purpose of this.

Arkatos1 replied:
It sets a color of the Blob marker when the Blob first spawns and it's reagent is randomly rolled. I added complementary colors, because if marker used the same color as the rest of the Blob, it would be hard to see for the player.

Fox-McCloud commented on code/modules/reagents/chemistry/reagents.dm:
```diff
@@ -8,6 +8,7 @@
 	var/volume = 0
 	var/metabolization_rate = REAGENTS_METABOLISM
 	var/color = "#000000" // rgb: 0, 0, 0 (does not support alpha channels - yet!)
+	var/complementary_color = "#000000"
```
This shouldn't be a reagent level var. It's only blob reagents that use it.

Arkatos1 replied:
I moved the var where it belongs. Fixed.
